pkg/datarepair/checker/checker.go

```diff
@@ -127,8 +141,19 @@ func (checker *Checker) IdentifyInjuredSegments(ctx context.Context) (err error)
 
 					// reset durability stats for next iteration
 					checker.monStats = durabilityStats{}
+
+					if checker.pieceTracker != nil {
```

Comment:
This is called inside defer, which may end the checker loop prematurely with error. In that case we shouldn't send since it could end up wiping all data.